File: DataCollector/src/test/java/com/braintreepayments/api/datacollector/DataCollectorInternalRequestUnitTest.java
package com.braintreepayments.api.datacollector;

import static junit.framework.Assert.assertEquals;

import org.junit.Test;

public class DataCollectorInternalRequestUnitTest {

    @Test
    public void setClientMetadataId_trimsId_to_32characters() {
        DataCollectorInternalRequest request = new DataCollectorInternalRequest(true)
                .setRiskCorrelationId("pairing-id-pairing-id-pairing-idXXX");

        assertEquals("pairing-id-pairing-id-pairing-id", request.getClientMetadataId());
        assertEquals(32, request.getClientMetadataId().length());
    }
}

File: DataCollector/src/test/java/com/braintreepayments/api/datacollector/DataCollectorInternalRequestUnitTest.kt
package com.braintreepayments.api.datacollector

import org.junit.Assert
import org.junit.Test

class DataCollectorInternalRequestUnitTest {
    @Test
    fun `when clientMetadataId is set with a value longer than 32 characters, it is trimmed to 32 characters`() {
        val request = DataCollectorInternalRequest(true)
        request.clientMetadataId = "pairing-id-pairing-id-pairing-idXXX"

        Assert.assertEquals("pairing-id-pairing-id-pairing-id", request.clientMetadataId)
        Assert.assertEquals(32, request.clientMetadataId!!.length)
    }
}
